sponsorship_compassion/view/contract_view.xml

```diff
@@ -1,139 +1,34 @@
 			<field name="arch" type="xml">
-				<data>
-					<tree string="Contracts" position="attributes">
-						<attribute name="colors">blue:state == 'draft';red:state == 'mandate';green:state == 'waiting';black:state == 'active';gray:state in ('terminated','cancelled')</attribute>
-						<attribute name="string">Sponsorships</attribute>
-					</tree>
-					<field name="partner_id" position="after">
-						<field name="child_code" />
-						<field name="child_name" />
-					</field>
-					<field name="partner_id" position="before">
-						<field name="partner_codega" invisible="context.get('partner_contracts_view')" />
-					</field>
-                    <field name="payment_term_id" position="after">
-                        <field name="frequency" />
-                        <field name="origin_id" />
-                    </field>
-                    <field name="state" position="after">
-                        <button name="open_contract" type="object" invisible="not context.get('partner_contracts_view')" icon="STOCK_JUMP_TO" />
-                    </field>
-					<field name="reference" position="replace" />
-				</data>
-			</field>
+                <button name="contract_terminated" position="after">
+                     <button name="%(action_end_sponsorship)d" type="action" string="Terminate" class="oe_highlight" attrs="{'invisible': ['|',('type', '!=', 'S'),('state', 'not in', ['waiting', 'mandate', 'active'])]}"/>
+                </button>          
+            </field>
 		</record>
-		
-		<record id="view_recurring_contract_form_compassion" model="ir.ui.view">
-			<field name="name">recurring.contract.compassion.form</field>
-			<field name="model">recurring.contract</field>
-			<field name="inherit_id" ref="recurring_contract.view_recurring_contract_form"/>
-			<field eval="18" name="priority"/>
-			<field name="arch" type="xml">
-                <!-- Modify header to add the waiting state -->
-                <field name="state" position="attributes">
-                    <attribute name="statusbar_visible">draft,active,waiting,terminated</attribute>
-                </field>
 
-                <!-- Replace terminate button to call the end_sponsorship wizard -->
-                <button name="contract_terminated" position="replace">
-                     <button name="%(action_end_sponsorship)d" states="waiting,mandate,active" type="action" string="Terminate" class="oe_highlight" />
-                </button>
+        <record id="action_sponsorship_contract" model="ir.actions.act_window">
+            <field name="name">Sponsorships</field>
+            <field name="type">ir.actions.act_window</field>
+            <field name="res_model">recurring.contract</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">tree,form</field>
+            <field name="domain">[('type','=','S')]</field>
+            <field name="context">{'default_type':'S', 'product_type':['Sponsor gifts']}</field>

```

Comment:
I don't understand what is product_type. Is it a value not allowed in products ?